### PR TITLE
Separate api keys and connection notices

### LINKED ISSUE #438: Separate the connectivity check from the api keys check
## Expected behavior and actual behavior.
When the plugin can't connect to ebanx servers on admin pages during checks it should display a connection error if the problem is a conectivity one. When the keys are invalid, the pay api responds with something, so we say the keys are invalid instead of complaining of connectivity.

Using a single error message for 2 different situations is confusing the merchant.

## Steps to reproduce the problem.
Use bad api keys and save.

## Specifications like the version of the project, operating system, or hardware.
1.10.1

### PR DESCRIPTION
**Pull request by SparK-Cruz**
Closes #438

### CODE REVIEW

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php [hunk 1]
```diff
@@ -28,55 +28,152 @@ public static function check_sandbox_mode($context) {
...
+		\Ebanx\Config::set(array('integrationKey' => $context->private_key, 'testMode' => $context->is_sandbox_mode));
+		try {
 			$res = \Ebanx\Ebanx::getMerchantIntegrationProperties(array('integrationKey' => $context->private_key));
-			$res_public = \Ebanx\Ebanx::getMerchantIntegrationPublicProperties(array('public_integration_key' => $context->public_key));
+			if ($res->status === 'SUCCESS') {
+				return true;
+			}
+
+			$message = sprintf(
+				'EBANX - Your %s private integration key is invalid, please <a href="%s">adjust your settings</a>.',
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php]: Please, don't forget the method for translation `__`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php [hunk 2]
```diff
@@ -28,55 +28,152 @@ public static function check_sandbox_mode($context) {
...
+		\Ebanx\Config::set(array('integrationKey' => $context->private_key, 'testMode' => $context->is_sandbox_mode));
+		try {
+			$res = \Ebanx\Ebanx::getMerchantIntegrationPublicProperties(array('public_integration_key' => $context->public_key));
+			if ($res->status === 'SUCCESS') {
+				return true;
+			}
 
-			$message = sprintf('EBANX - Could not connect to our servers. Please check if your server can reach our API (<a href="%1$s">%1$s</a>) and your integrations keys are correct.', $api_url);
+			$message = sprintf(
+				'EBANX - Your %s public integration key is invalid, please <a href="%s">adjust your settings</a>.',
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php]: Please, don't forget the method for translation `__`

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php [hunk 3]
```diff
@@ -28,55 +28,152 @@ public static function check_sandbox_mode($context) {
...
+		catch (RuntimeException $e) {
+			self::connection_error($context);
+		}
+		return false;
+	}
+
+	private static function connection_error($context) {
+		$api_url = 'https://api.ebanx.com';
+
+		$message = sprintf('EBANX - Could not connect to our servers. Please check if your server can reach our API (<a href="%1$s">%1$s</a>).', $api_url);
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php]: Please, use the method for translation `__`

**Comment by SparK-Cruz** [comment]
Next step will be to decouple these two checks, but that would require a "ping" route in Pay API

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php [hunk 4]
```diff
@@ -101,8 +101,8 @@ public static function check_private_key($context) {
 			}
 
 			$message = sprintf(
-				__('EBANX - Your <strong>%s integration key</strong> is invalid, please <a href="%s">adjust your settings</a>.', 'woocommerce-gateway-ebanx'),
-				$context->is_sandbox_mode ? "sandbox" : "live",
+				__('EBANX - Your <strong>%s Integration Key</strong> is invalid, please <a href="%s">adjust your settings</a>.', 'woocommerce-gateway-ebanx'),
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-checker.php]: And when is sandbox mode? What could we do?

**Pull request merged**